INTMDB-364 Add support for serverless private endpoints (AWS + Azure) (#913)
mongodb/terraform-provider-mongodbatlas: website/docs/d/privatelink_endpoints_service_serverless.html.markdown

}

resource "mongodbatlas_serverless_instance" "test" {
	project_id   = "<PROJECT_ID>"
	name         = "test-db"
	provider_settings_backing_provider_name = "AWS"
	provider_settings_provider_name = "SERVERLESS"
	provider_settings_region_name = "US_EAST_1"
	continuous_backup_enabled = true
}
```

## Example with AZURE
```terraform

data "mongodbatlas_privatelink_endpoints_service_serverless" "test" {
  project_id   = "<PROJECT_ID>"
  instance_name = mongodbatlas_serverless_instance.test.name
}

resource "mongodbatlas_privatelink_endpoint_serverless" "test" {
	project_id   = "<PROJECT_ID>"
	instance_name = mongodbatlas_serverless_instance.test.name
	provider_name = "AZURE"
}
	  
resource "mongodbatlas_privatelink_endpoint_service_serverless" "test" {
	project_id   = "<PROJECT_ID>"
	instance_name = "test-db"
	endpoint_id = mongodbatlas_privatelink_endpoint_serverless.test.endpoint_id
	provider_name = "AZURE"
	comment = "New serverless endpoint"
}

resource "mongodbatlas_serverless_instance" "test" {
	project_id   = "<PROJECT_ID>"
	name         = "test-db"
	provider_settings_backing_provider_name = "AZURE"
	provider_settings_provider_name = "SERVERLESS"

> 	provider_settings_region_name = "US_EAST_1"

Reviewer: Azure doesn't have a US_EAST_1.  
```suggestion
	provider_settings_region_name = "US_EAST"
```

	continuous_backup_enabled = true
}
```

## Argument Reference

* `project_id` - (Required) Unique 24-digit hexadecimal string that identifies the project.
* `instance_name` - Human-readable label that identifies the serverless instance

## Attributes Reference

In addition to all arguments above, the following attributes are exported:
* `results` - Each element in the `result` array is one private serverless endpoint.

### results

Each object in the `results` array represents an online archive with the following attributes:
* `cloud_provider_endpoint_id` - Unique string that identifies the private endpoint's network interface.
* `comment` - Human-readable string to associate with this private endpoint.

> * `endpoint_id` - (Required) Unique 22-character alphanumeric string that identifies the private endpoint. Atlas supports AWS private endpoints using the [|aws| PrivateLink](https://aws.amazon.com/privatelink/) feature.

Reviewer 2: ```suggestion
* `endpoint_id` - (Required) Unique 22-character alphanumeric string that identifies the private endpoint. Atlas supports AWS private endpoints using the [AWS PrivateLink](https://aws.amazon.com/privatelink/) feature.
```

---
layout: "mongodbatlas"
page_title: "MongoDB Atlas: privatelink_endpoints_service_serverless"
sidebar_current: "docs-mongodbatlas-datasource-privatelink-endpoints-service-serverless"
description: |-
Describes the list of all Serverless PrivateLink Endpoint Service
---

# Data Source: privatelink_endpoints_service_serverless

`privatelink_endpoints_service_serverless` Describes the list of all Serverless PrivateLink Endpoint Service resource.

-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.

## Example Usage

### Basic
```terraform

data "mongodbatlas_privatelink_endpoints_service_serverless" "test" {
  project_id   = "<PROJECT_ID>"
  instance_name = mongodbatlas_serverless_instance.test.name
}

resource "mongodbatlas_privatelink_endpoint_serverless" "test" {
	project_id   = "<PROJECT_ID>"
	instance_name = mongodbatlas_serverless_instance.test.name
	provider_name = "AWS"
}

> 	  

Reviewer: same comment as above on full example, with AWS side

Reviewer: Resolving as not as important on data source side and we can fix later.

resource "mongodbatlas_privatelink_endpoint_service_serverless" "test" {
	project_id   = "<PROJECT_ID>"
	instance_name = "test-db"
	endpoint_id = mongodbatlas_privatelink_endpoint_serverless.test.endpoint_id
	provider_name = "AWS"
	comment = "New serverless endpoint"
}

resource "mongodbatlas_serverless_instance" "test" {
	project_id   = "<PROJECT_ID>"
	name         = "test-db"
	provider_settings_backing_provider_name = "AWS"
	provider_settings_provider_name = "SERVERLESS"
	provider_settings_region_name = "US_EAST_1"
	continuous_backup_enabled = true
}
```

## Argument Reference

* `project_id` - (Required) Unique 24-digit hexadecimal string that identifies the project.

> * `instance_name` - (Required) Serverless instance name to attach private

Reviewer: ```suggestion
* `instance_name` - (Required) Human-readable label that identifies the serverless instance 
```

`privatelink_endpoints_service_serverless` Describes the list of all Serverless PrivateLink Endpoint Service resource.

-> **NOTE:** Groups and projects are synonymous terms. You may find group_id in the official documentation.

## Example Usage

### Basic
```terraform

data "mongodbatlas_privatelink_endpoints_service_serverless" "test" {
  project_id   = "<PROJECT_ID>"
  instance_name = mongodbatlas_serverless_instance.test.name
}

resource "mongodbatlas_privatelink_endpoint_serverless" "test" {
	project_id   = "<PROJECT_ID>"
	instance_name = mongodbatlas_serverless_instance.test.name
	provider_name = "AWS"
}
	  
resource "mongodbatlas_privatelink_endpoint_service_serverless" "test" {
	project_id   = "<PROJECT_ID>"
	instance_name = "test-db"
	endpoint_id = mongodbatlas_privatelink_endpoint_serverless.test.endpoint_id
	provider_name = "AWS"
	comment = "New serverless endpoint"
}

resource "mongodbatlas_serverless_instance" "test" {
	project_id   = "<PROJECT_ID>"
	name         = "test-db"
	provider_settings_backing_provider_name = "AWS"
	provider_settings_provider_name = "SERVERLESS"
	provider_settings_region_name = "US_EAST_1"
	continuous_backup_enabled = true

> }

Reviewer 2: Missing example with AZURE
